exporter/opentelemetry-exporter-otlp-proto-grpc/src/opentelemetry/exporter/otlp/proto/grpc/exporter.py

```diff
@@ -130,41 +133,16 @@ def _translate_key_values(key: str, value: Any) -> KeyValue:
     return KeyValue(key=key, value=_translate_value(value))
 
 
+@deprecated(
+    version="1.16.0",
+    reason="Use one of the encoders from opentelemetry-exporter-otlp-proto-common instead",
+)
 def get_resource_data(
```

Comment:
@srikanthccv This is how I'd deprecate, I'd also do something similar for the one that was [here](https://github.com/open-telemetry/opentelemetry-python/pull/3169/files/1e7f6b101abe0c160101ce6d4be47945baafa16e#diff-dd984a10301c8b38d590d7ebb358d9f51846d986cdbbd9e1c9e785ca18852ebcL409) issue is this has a slightly different behavior as it supports mappings. I've raised the mappings question [here](https://github.com/open-telemetry/opentelemetry-python/pull/3169#discussion_r1101188361).

Reply:
We can add the support for mapping separately.

Reply:
Ok, keep the two implementations of `get_resource_data` one that supports mappings and one that doesn't—keeping it so metrics include mappings, with logs and traces not having mappings? This goes for both the `grpc` and `http` packages. Keeping these mappings/no mappings seems a bit convoluted to me, as we already have something with mappings and something without